src/main/java/crazypants/enderio/machine/solar/SolarPanelNetwork.java

```diff
@@ -61,21 +89,53 @@ boolean addToNetwork(TileEntitySolarPanel panel) {
     return true;
   }
 
+  void cleanupMemberlist() {
+    if (!panels.isEmpty()) {
+      int energyPerPanel = energy.getEnergyStored() / panels.size();
+      Iterator<BlockCoord> iterator = panels.iterator();
+      while (iterator.hasNext()) {
+        BlockCoord panel = iterator.next();
+        boolean isGood = false;
+        if (world.blockExists(panel.x, panel.y, panel.z)) {
+          TileEntity tileEntity = panel.getTileEntity(world);
+          if (tileEntity instanceof TileEntitySolarPanel) {
+            if (((TileEntitySolarPanel) tileEntity).network == this) {
+              isGood = true;
+            }
+          }
+        }
+        if (!isGood) {
```

Comment:
Could the variable be isBad or similiar instead? the ! is easy to miss when reading